freva-rest/src/freva_rest/databrowser_api/core.py

```diff
@@ -980,28 +981,47 @@ async def extended_search(
         self.query["facet.field"] = self.translator.translate_facets(
             search_facets, backwards=True
         )
+
         self.query["fl"] = [self.uniq_key, "fs_type"]
         logger.info(
             "Query %s for uniq_key: %s with %s",
             self.url,
             self.uniq_key,
             self.query,
         )
-
         async with self._session_get() as res:
             search_status, search = res
+
+        docs = search.get("response", {}).get("docs", [])
+
+        if zarr_stream and docs:
+            api_path = f"{self._config.proxy}/api/freva-nextgen/data-portal/zarr"
+            for doc in docs:
+                uri = doc[self.uniq_key]
+                uuid5 = str(uuid.uuid5(uuid.NAMESPACE_URL, uri))
+                try:
+                    cache = await create_redis_connection()
+                    await cache.publish(
+                        "data-portal",
+                        json.dumps(
+                            {"uri": {"path": uri, "uuid": uuid5}}
+                        ).encode("utf-8"),
+                    )
+                    doc[self.uniq_key] = f"{api_path}/{uuid5}.zarr"
+                except Exception as pub_err:
+                    logger.error(
+                        "Failed to publish to Redis for %s: %s", uri, pub_err
+                    )
+                    doc[self.uniq_key] = (
+                        "Internal error, service not able to publish"
+                    )
+                doc["fs_type"] = doc.get("fs_type", "posix")
```

Comment:
How about using an external func for this? Then you can use this external func here and in the data-portal endpoint.